src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3388,6 +3391,21 @@ private int MinBatchSize
             }
         }
 
+        private Size? ManagedAllocationLimit
+        {
+            get
+            {
+                if (_managedAllocationLimit != null)
+                    return _managedAllocationLimit.Value;
+
+                var limit = Configuration.MangedAllocationsBatchLimit;
+                
+                _managedAllocationLimit = new Lazy<Size?>(() => limit);
+
+                return _managedAllocationLimit.Value;
+            }
+        }
+
```

Comment:
Why? There is nothing expensive here that we are trying to save

Reply:
this is how it is done for other stuff:
https://github.com/ravendb/ravendb/blob/8379818ad7c937cb4393321886749631741074ce/src/Raven.Server/Documents/Indexes/Index.cs#L3591

Reply:
Let's just init all of that normally in the ctor

Reply:
we don't have the configuration in the ctor
I will read it directly from the config `Configuration.MangedAllocationsBatchLimit`